upgrade/v2.5/script/upgrade.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/sh
+
+# Set the ENV var for KDD backend
+export DATASTORE_TYPE=kubernetes
+
+# Create the CRDs by applying the crds.yaml manifest.
+echo "Creating Calico CRDs..."; echo
+/sbin/kubectl apply -f crds.yaml
+/sbin/kubectl apply -f tprs.yaml
+
+# Get IPPools list with old calicoctl (from TPR) and save it as a yaml file.
+# Apply the yaml file with the new calicoctl (to CRD)
+echo "Migrating IPPools..."
+/sbin/calicoctl-v1.4 get ippool -o yaml > ippool.yaml
+if [ $? != 0 ]; then
+    echo "Failed to get IPPools throught calicoctl"
+    exit 1
+else if [ `cat ippool.yaml | wc -l` == 1 ]; then
+    echo "No IPPools found to migrate."
+    echo "Moving on..."
+else
+    echo "Migrating the following IPPools..."
+    cat ippool.yaml
+    echo
+    /sbin/calicoctl-v1.5 apply -f ippool.yaml
+    if [ $? != 0 ]; then
+        echo "Failed to apply the IPPools"
+        exit 1
+    else
+        echo "Successfully migrated the IPPools!"
+    fi
+  fi
+fi
+
+# Get BGPPeers list with old calicoctl (from TPR) and save it as a yaml file.
+# Apply the yaml file with the new calicoctl (to CRD)
+echo; echo "Migrating BGPPeers..."
+/sbin/calicoctl-v1.4 get bgppeers -o yaml > bgppeers.yaml
+if [ $? != 0 ]; then
+    echo "Failed to get Global BGP Peers through calicoctl"
+    exit 1
+else if [ `cat ippool.yaml | wc -l` == 1 ]; then
+    echo "No BGPPeers found to migrate."
+    echo "Moving on..."
+else
+    echo "Migrating the following BGPPeers..."
+    cat bgppeers.yaml
+    echo
+    /sbin/calicoctl-v1.5 apply -f bgppeers.yaml
+    if [ $? != 0 ]; then
+        echo "Failed to apply the BGPPeers"
+        exit 1
+    else
+        echo "Successfully migrated the BGPPeers!"
+    fi
+  fi
+fi
+
+# List all the Felix configs using kubectl and save it in a yaml file.
+# Change the apiVersion from 'projectcalico.org/v1' to 'crd.projectcalico.org/v1',
+# rename resource kind from 'GlobalConfig' to 'GlobalFelixConfig' and save it in a new yaml file.
+# Apply the modified yaml file.
+echo; echo "Migrating GlobalFelixConfig..."
+/sbin/kubectl get globalconfig --all-namespaces -o yaml > tpr-felixconfig.yaml
+if [ $? != 0 ]; then
+    echo "Failed to get Global Felix config through Kubectl"
+    exit 1
+else
+    echo "Successfully got the Felix Config:"
+    cat tpr-felixconfig.yaml
+    cat tpr-felixconfig.yaml | grep 'items\: \[\]'
+    if [ $? == 0 ]; then
+        echo "No GlobalFelixConfig found to migrate."
+        echo "Moving on..."
+    else
+        cat tpr-felixconfig.yaml | sed '/apiVersion/s/projectcalico\.org\/v1/crd\.projectcalico\.org\/v1/g' | sed '/kind/s/GlobalConfig/GlobalFelixConfig/g' > crd-felixconfig.yaml
+        echo "Successfully renamed the resource kind"
```

Comment:
and the apiGroup.

Perhaps just say you converted from `blah.v1/MyTPR` to `crd.blah.v1/MyNewFunkyCRD` - that way the log is a little more useful.